### ISSUE TITLE
"Follow in dump" missing in arguments view

### ISSUE THREAD

**Issue by justanotheranonymoususer** [x64dbg/x64dbg#3229]
### Operating System

Win11

### x64dbg Version

Aug 1 2023

### Describe the issue

The one item that I needed was missing, not sure if intentional.

In general I find the "follow" commands of x64dbg confusing, for example when click on a command with indirection, but maybe it's just me.

![image](https://github.com/x64dbg/x64dbg/assets/5781692/c2ac02b8-20d1-4a75-8445-5251ef99a40e)

### Steps to reproduce

1. 
2. 
3.

**Comment by mrexodia** [comment]
Yeah this is a bug and additionally these things are confusingly labeled too...

**Comment by HimaTeju** [comment]
Can I get some description on how to approach to solve this?

**Comment by mrexodia** [comment]
The logic for the argument widget is handled here: https://github.com/x64dbg/x64dbg/blob/development/src/gui/Src/Gui/CPUArgumentWidget.cpp#L128-L174

**Comment by HimaTeju** [comment]
Thank you! @mrexodia

**Comment by foralost** [comment]
@HimaTeju 
are you working on this? If not, I could try to fix it.

**Comment by foralost** [comment]
I will work on it 25-26.11.

**Comment by foralost** [comment]
Well, I see that this is already solved:
https://github.com/x64dbg/x64dbg/blob/development/src/gui/Src/Gui/CPUArgumentWidget.cpp#L153

![image](https://github.com/x64dbg/x64dbg/assets/28221162/fb9a5e68-e5b5-4b31-b9df-4e42a25f7219)

@mrexodia close?

**Comment by mrexodia** [comment]
I think it’s still confusing, even if you can technically follow it. The main issue is with the stack arguments, where I don’t think the user really need more than follow in stack for those at the end of the list.

**Comment by foralost** [comment]
Ok, so no `Follow XXX in Disassembler` and `Follow XXX in Dump`, only `Follow XXX in Stack` for stack arguments (number > 4 as stated in fastcall).

**Comment by mrexodia** [comment]
Hm I actually have to confirm this locally. Will get back to you!

**Comment by mrexodia** [comment]
I didn't get back to you 😅 

![Image](https://github.com/user-attachments/assets/9adf707d-a282-452a-a1ef-6218a7958ce0)

The issue here is that when there is a stack address pushed as an argument on the stack that you get very confusing context menus. We definitely do not want to follow a stack address (or any non-executable address) in the disassembler, so that menu can be removed.

The `8EE20` should say 'Show argument N on the stack' and I think similarly it would be clearer to say 'Follow rcx (0x12345) in dump' for the arguments themselves. This should also be at the bottom, because that is a very uncommon operation.

**Comment by d2k2-git** [comment]
for the first refactoring step check if the value is in NOT in the stack range, then only allow to follow it in the disassembler. the labeling of the context menu items is a cherry on top, since you can see the selected line which you clicked.

<img width="1011" height="320" alt="Image" src="https://github.com/user-attachments/assets/431ecc08-6505-4112-9af0-04d52dac3e93" />
